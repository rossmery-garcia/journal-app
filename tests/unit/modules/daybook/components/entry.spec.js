import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Entry component', () => {

  const mockRouter = {
    push: jest.fn()
  }

  const wrapper = shallowMount( Entry, {
    props: {
      entry: journalState.entries[0]
    },
    global: {
      mocks: {
        $router: mockRouter
      }
    }
  })

  test('Should match with the snapshot', () => {
    expect( wrapper.html() ).toMatchSnapshot()
  });

  test('should redirect when clicking on entry-container', () => {

    const entryContainer = wrapper.find('.entry-container')
    entryContainer.trigger('click')

    expect( mockRouter.push ).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: journalState.entries[0].id
      }
    })
  })

  test('computed properties', () => {

    expect( wrapper.vm.day).toBe(8)
    expect( wrapper.vm.month).toBe('Noviembre')
    expect( wrapper.vm.yearDay).toBe('2022, Martes')
  })
})
