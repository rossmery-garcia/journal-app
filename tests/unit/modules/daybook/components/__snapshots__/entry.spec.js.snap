// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component Should match with the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">8</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-light">2022, Martes</span></div>
  <div class="entry-description">First entry from mock-data</div>
</div>
`;
